components/table/src/table-style.service.ts

```diff
@@ -31,8 +31,21 @@ export class NzTableStyleService {
     this.manualWidthConfigPx$,
     combineLatest([this.listOfAutoWidthPx$, this.manualWidthConfigPx$]).pipe(
       map(([autoWidth, manualWidth]) => {
-        /** use autoWidth until column length match **/
-        return autoWidth.length !== manualWidth.length ? manualWidth : autoWidth;
+        let realManualWidthLength = 0;
+        for (const width of manualWidth) {
+          if (width) {
+            realManualWidthLength += 1;
+          }
+        }
+        // If manualWidth is fully filled with width and the length is equal to autoWidth, regard autoWidth as the result.
+        if (realManualWidthLength > 0 && realManualWidthLength === autoWidth.length) {
```

Comment:
Hi, this won't fix this issue since realManualWidthLength will equal autoWidth.length when update manualWidthConfigPx manually

Reply:
Thank you for your advice.  I'll try to fix this.